BLD: having difficulties building with more modern rtools versions

At the moment we use the fairly old `rtools40` toolchain to build [scipy on windows](https://github.com/scipy/scipy/blob/main/.github/workflows/windows.yml#L80).
In https://github.com/andyfaff/scipy/pull/91 I'm trying to see if we can bump up to a more recent version of rtools (rtools43/rtools44/rtools45). I thought it would be a simple case of changing:
```
choco install rtools -y --no-progress --force --version=4.0.0.20220206
echo "c:\rtools40\ucrt64\bin;" >> $env:GITHUB_PATH
```
to something like:
```
choco install rtools -y --no-progress --force --version=<more recent version>
echo "c:\rtools45\ucrt64\bin;" >> $env:GITHUB_PATH
```

One of the benefits of rtools45 is that it already comes installed on the GH images. However, I'm experiencing problems.
See https://github.com/andyfaff/scipy/actions/runs/22925715033/job/66535458690 for example, which was trying rtools43. At the tail end of the [build log](https://github.com/andyfaff/scipy/actions/runs/22925715033/job/66535458690#step:6:1950) there are whole bunch of undefined symbols (e.g. `undefined reference to `__imp_ccoshl'`).

I did manage to get things to build once, 58e39933624db7d000f0c413332b127fcb8966cf, which used the preinstalled rtools45:
```
echo "c:\rtools45\ucrt64\bin;C:\rtools45\x86_64-w64-mingw32.static.posix\bin;C:\rtools45\usr\bin" >> $env:GITHUB_PATH
```
but got all the test workers [crashing immediately](https://github.com/andyfaff/scipy/actions/runs/22899657108/job/66442711534#step:7:49). If I had to guess I would suggest the wrong runtime was used.

Does anyone have any insights as to how I can get things working?
It's not like rtools40 is going to go away soon, but it does make the whole build process more robust and future-proof if a more modern toolset can be used..

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -73,12 +73,11 @@ jobs:
         with:
           python-version: 3.12
 
-      - name: win_amd64 - install rtools
+      - name: win_amd64 - configure rtools
         if: ${{ matrix.buildplat[1] == 'win' && matrix.buildplat[2] == 'AMD64' }}
         run: |
           # mingw-w64
-          choco install rtools -y --no-progress --force --version=4.0.0.20220206
-          echo "c:\rtools40\ucrt64\bin;" >> $env:GITHUB_PATH
+          echo "c:\rtools45\ucrt64\bin;" >> $env:GITHUB_PATH
 
       - name: Set environment variables for ARM64
         if: matrix.buildplat[1] == 'win' && matrix.buildplat[2] == 'ARM64'

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -74,15 +74,14 @@ jobs:
           cache: "pip"
           cache-dependency-path: "environment.yml"
 
-      - name: Install rtools (mingw-w64)
+      - name: Set rtools location location (mingw-w64)
         run: |
-          choco install rtools -y --no-progress --force --version=4.0.0.20220206
-          echo "c:\rtools40\ucrt64\bin;" >> $env:GITHUB_PATH
+          echo "c:\rtools45\ucrt64\bin" >> $env:GITHUB_PATH
 
       - name: pip-packages
         run: |
           # 2.0.0 is currently our oldest supported NumPy version
-          python -m pip install numpy==2.0.0 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pooch spin hypothesis "click<8.3.0"
+          python -m pip install numpy==2.0.0 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pooch spin hypothesis "click<8.3.0" pkgconf
           python -m pip install -r requirements/openblas.txt
 
       - name: Build

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -74,7 +74,7 @@ jobs:
           cache: "pip"
           cache-dependency-path: "environment.yml"
 
-      - name: Set rtools location location (mingw-w64)
+      - name: Set rtools location (mingw-w64)
         run: |
           echo "c:\rtools45\ucrt64\bin" >> $env:GITHUB_PATH
 

--- a/doc/source/dev/toolchain.rst
+++ b/doc/source/dev/toolchain.rst
@@ -153,7 +153,7 @@ Currently, SciPy wheels are being built as follows:
  OSX x86_64 (Accelerate)     ``macos-15-intel``               Apple clang 15.0.0/gfortran 13.2.0     ``cibuildwheel``
  OSX arm64 (OpenBLAS)        ``macos-14``                     Apple clang 15.0.0/gfortran 12.1.0     ``cibuildwheel``
  OSX arm64 (Accelerate)      ``macos-14``                     Apple clang 15.0.0/gfortran 13.2.0     ``cibuildwheel``
- Windows                     ``windows-2019``                 GCC 10.3.0 (`rtools`_)                 ``cibuildwheel``
+ Windows                     ``windows-2025``                 GCC 10.3.0 (`rtools`_)                 ``cibuildwheel``
 =========================   ==============================   ====================================   =============================
 
 .. _CI: https://github.com/actions/runner-images

--- a/doc/source/dev/toolchain.rst
+++ b/doc/source/dev/toolchain.rst
@@ -153,7 +153,7 @@ Currently, SciPy wheels are being built as follows:
  OSX x86_64 (Accelerate)     ``macos-15-intel``               Apple clang 15.0.0/gfortran 13.2.0     ``cibuildwheel``
  OSX arm64 (OpenBLAS)        ``macos-14``                     Apple clang 15.0.0/gfortran 12.1.0     ``cibuildwheel``
  OSX arm64 (Accelerate)      ``macos-14``                     Apple clang 15.0.0/gfortran 13.2.0     ``cibuildwheel``
- Windows                     ``windows-2025``                 GCC 10.3.0 (`rtools`_)                 ``cibuildwheel``
+ Windows                     ``windows-2025``                 GCC 15.2.0 (`rtools`_)                 ``cibuildwheel``
 =========================   ==============================   ====================================   =============================
 
 .. _CI: https://github.com/actions/runner-images

--- a/requirements/pkgconf.txt
+++ b/requirements/pkgconf.txt
@@ -1,0 +1,1 @@
+pkgconf==2.5.1.post1

--- a/tools/wheels/cibw_before_build.sh
+++ b/tools/wheels/cibw_before_build.sh
@@ -73,4 +73,7 @@ fi
 # cibuildwheel doesn't install delvewheel by default
 if [[ $RUNNER_OS == "Windows" ]]; then
     python -m pip install -r $PROJECT_DIR/requirements/delvewheel_requirements.txt
+    # pkgconf - carries out the role of pkg-config.
+    # Alternative is pkgconfiglite that you have to install with choco
+    python -m pip install -r $PROJECT_DIR/requirements/pkgconf.txt
 fi

